modules/auxiliary/analyze/pcap_ot_iot_analyzer.rb

```diff
@@ -0,0 +1,218 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'PCAP Analyzer for OT/IOT Devices',
+        'Description' => %q{
+          This module analyzes .pcap files to extract network information and identify OT/IOT devices.
+          It generates a CSV output file with the extracted data.
+        },
+        'Author' => ['Jordi Ubach'],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['Email', 'jordiubach@protonmail.com']
+        ]
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('FILE_PCAP', [true, 'Path to the .pcap file to analyze']),
+        OptString.new('FILE_PORT', [true, 'Path to the .txt file with port/protocol information']),
+        OptString.new('FILE_OUT', [true, 'Path to the output .csv file'])
+      ]
+    )
+  end
+
+  def run
+    # Verify input parameters
+    unless File.exist?(datastore['FILE_PCAP']) && File.extname(datastore['FILE_PCAP']) == '.pcap'
+      print_error("Invalid or non-existent PCAP file: #{datastore['FILE_PCAP']}")
+      return
+    end
+
+    unless File.exist?(datastore['FILE_PORT']) && File.extname(datastore['FILE_PORT']) == '.txt'
+      print_error("Invalid or non-existent port file: #{datastore['FILE_PORT']}")
+      return
+    end
+
+    if File.exist?(datastore['FILE_OUT'])
+      print_error("Output file already exists: #{datastore['FILE_OUT']}")
+      return
+    end
```

Comment:
Instead of print_error for each of these you should use `fail_with(Failure::BadConfig, ...`